tests/Main.fs

```diff
@@ -148,8 +152,26 @@ describe "Thoth.Fetch" <| fun _ ->
             }
             |> Promise.catch d
             |> Promise.start
+
+        it "Fetch.fetchAs works with extra coder" <| fun d ->
+            promise {
+                let! res = Fetch.fetchAs("http://localhost:3000/books/1", extra = bookCoder, isCamelCase = true)
```

Comment:
This is weird.
If `isCamelCase` is set to `false`, not only this single test fails but 27. 
IMHO it should not fail in the first case, as an encoder is provided. The other test fail because of caching?

Reply:
* It failed in the first test because the extra coder was not passed to tryFetchAs :-(
* Yes, it is a caching issue. Caching within Thoth.Json doesn't include the casing within the key for the cached item. As the extra coder was not passed, the Auto decoder was cached for isCamelCase = false. And json server was configured for camelCase.

Reply:
Sorry for the delay, I was being sick.

Yes like you discovered, this is a limitation from current Thoth.Json mechanism. I am going to merge your work on Thoth.Json before as it's needed for this PR to work I think :)